mne/io/proj.py

```diff
@@ -108,7 +108,10 @@ def add_proj(self, projs, remove_existing=False, verbose=None):
     def add_eeg_average_proj(self):
         """Add an average EEG reference projector if one does not exist
         """
-        if _needs_eeg_average_ref_proj(self.info):
+        if self.info['custom_ref_applied']:
+            logger.warning('A custom reference was previously applied, '
+                           'average reference cannot be created.')
```

Comment:
Turns out it wasn't my idea ;) `_needs_eeg_average_ref_proj` already silently returns `False` if custom ref applied! So someone else must have decided that should be the case. I added a warning to `add_eeg_average_proj` that nothing happens if `info['custom_ref_applied']`, otherwise it's pretty confusing.